### PR TITLE
feat: press cmd+f to focus on the search bar

### PR DESCRIPTION
**Pull request by ChasLui**

### CODE REVIEW

File: web/src/components/SearchBar.tsx [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
+    }
+  };
 
   useEffect(() => {
     const text = locationStore.getState().query.text;
     setQueryText(text === undefined ? "" : text);
+    document.body.addEventListener("keydown", handleKeyDown);
+    return () => {
+      document.body.removeEventListener("keydown", handleKeyDown);
+    };
```
Comment (boojack) [inline web/src/components/SearchBar.tsx:35]: separate the listener into another `useEffect`.

File: web/src/components/SearchBar.tsx [hunk 2]
```diff
@@ -1,19 +1,38 @@
...
+      return;
+    }
+    const isMetaKey = event.ctrlKey || event.metaKey;
+    if (isMetaKey && event.key === "f") {
+      if (dialogStore.getState().dialogStack.length) return;
+      event.preventDefault();
+      inputRef.current.focus();
+      return;
+    }
+  };
```
Comment (boojack) [inline web/src/components/SearchBar.tsx:27]: put this into `useEffect`

**Comment by ChasLui** [comment]
ci error

File: web/src/components/SearchBar.tsx [hunk 3]
```diff
@@ -1,15 +1,39 @@
...
+    if (isMetaKey && event.key === "f") {
+      event.preventDefault();
+      inputRef.current.focus();
+      return;
+    }
+  };
+
+  useEffect(() => {
+    document.body.addEventListener("keydown", handleKeyDown);
+    return () => {
```
Comment (boojack) [inline web/src/components/SearchBar.tsx:33]:
```suggestion
  useEffect(() => {
    const handleKeyDown = (event: KeyboardEvent) => {
      if (!inputRef.current) {
        return;
      }
      if (dialogStore.getState().dialogStack.length) {
        return;
      }
      const isMetaKey = event.ctrlKey || event.metaKey;
      if (isMetaKey && event.key === "f") {
        event.preventDefault();
        inputRef.current.focus();
        return;
      }
    };
    document.body.addEventListener("keydown", handleKeyDown);
    return () => {
```

**Pull request merged**